packages/shorebird_cli/lib/src/commands/upgrade_command.dart

```diff
@@ -0,0 +1,149 @@
+import 'dart:io';
+
+import 'package:mason_logger/mason_logger.dart';
+import 'package:path/path.dart' as p;
+import 'package:shorebird_cli/src/command.dart';
+
+/// {@template upgrade_command}
+/// `shorebird upgrade`
+/// A command which upgrades your copy of Shorebird.
+/// {@endtemplate}
+class UpgradeCommand extends ShorebirdCommand {
+  /// {@macro upgrade_command}
+  UpgradeCommand({required super.logger, super.runProcess});
+
+  @override
+  String get description => 'Upgrade your copy of Shorebird.';
+
+  static const String commandName = 'upgrade';
+
+  @override
+  String get name => commandName;
+
+  @override
+  Future<int> run() async {
+    final updateCheckProgress = logger.progress('Checking for updates');
+    final workingDirectory = p.dirname(Platform.script.toFilePath());
+
+    late final String currentVersion;
+    try {
+      currentVersion = await fetchCurrentVersion(
+        workingDirectory: workingDirectory,
+      );
+    } on ProcessException catch (error) {
+      updateCheckProgress.fail();
+      logger.err('Fetching current version failed: ${error.message}');
+      return ExitCode.software.code;
+    }
+
+    late final String latestVersion;
+    try {
+      latestVersion = await fetchLatestVersion(
+        workingDirectory: workingDirectory,
+      );
+    } on ProcessException catch (error) {
```

Comment:
You're going to delete so much code when you make this on ProcessException into some sort of function wrapper thing. 🤣

Reply:
Yeah we really need to do that 😅